.github/workflows/docs.yml

```diff
@@ -15,6 +26,66 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Setup Spack
+        uses: spack/setup-spack@v2
+        with:
+          ref: develop
+          buildcache: true
+          color: true
+          # This seems unnecessary, but otherwise we clone into ./spack
+          # This then would override the spack local repo dir, and break CI
+          path: spack-path
+
+      - name: Setup Environment
+        run: |
+          PALACE_SPEC=palace
+
+          # Spack.yaml with most / all settings configured
+          cat << EOF > spack.yaml
+          spack:
+            specs:
+              - ${PALACE_SPEC}
+            view: false
+            config:
+              install_tree:
+                root: /opt/spack
+                padded_length: False
+            concretizer:
+              reuse: false
+              unify: true
+              targets:
+                granularity: generic
+            packages:
+              petsc:
+                require: ~hdf5
+              rocblas:
+                require: ~tensile
+            repos:
+            - spack/local
+            mirrors:
+              spack:
+                binary: true
+                url: https://binaries.spack.io/${SPACK_VERSION}
+              local-buildcache:
+                binary: true
+                url: oci://${{ env.REGISTRY }}-${SPACK_VERSION}
+                signed: false
+                access_pair:
+                  id_variable: GITHUB_USER
+                  secret_variable: GITHUB_TOKEN
+          EOF
+
+      - name: Configure Binary Mirror Keys
+        run: |
+          # If we cached these, that would be faster and safer
+          spack -e . buildcache keys --install --trust
+
+      - name: Bootstrap Spack
+        run: spack  -e . bootstrap now
+
+      - name: Install Palace
+        run: spack  -e . install palace
+
```

Comment:
For reference

```
      # Push built binaries, even if the build fails
      # NOTE: This might fail as external fork PRs can't push to GHCR.
      - name: Push to GHCR cache
        if: |
          !cancelled() &&
          (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository)
        run: spack -e . buildcache push --force --with-build-dependencies --unsigned --update-index local-buildcache
```